main.cpp

```diff
@@ -210,285 +239,379 @@ struct Museum
     float maximumCurrentInAmps = 1.f;
-    // 5) Fuse state (bool)
     bool fuseState { true };
 
-    // 1) The mains input voltage can be switched on / off
     void setPowerState(bool powerState);
-    // 2) the Main output voltage can be switched on independently (standby)
     void setStandbyState(bool standbyState);
-    // 3) The mains fuse can blow
-    void blowFuse() { fuseState = false;}
+    void blowFuse(); //{ fuseState = false;}
+private:
+    bool _powerState = 1;
+    bool _standbyState = 1;
+    float powerTransformerMainRatio = 4.f;
+    float powerTransformerHeaterRatio = 0.055f;
 };
-/*
-Thing 6) Output Section
-5 properties:
-    1) the number of output tubes (int)
-    2) the grid bias voltage (float)
-    3) the output transformers primary impedance in  (double)
-    4) the output transformers secondary impedance in Ohms (float)
-    5) the maximum output power in watts (float)
-3 things it can do:
-    1) Warm up tubes
-    2) Admust master output volume
-    3) Amplify line level audio to a speaker level signal
- */
+
+void PowerSupply::setPowerState(bool powerState)
+{
+    _powerState = powerState;
+    if(fuseState)
+    {
+        if(_powerState)
+        {
+            // do power on things like heater on and if standby on
+            heaterOutputVoltage = mainsInputVoltage * powerTransformerHeaterRatio;
+            if (_standbyState)
+                mainOutputVoltage = mainsInputVoltage * powerTransformerMainRatio;
+        }
+        else
+        {
+            // do power off things
+            mainOutputVoltage = 0.f;
+            heaterOutputVoltage = 0.f;
+        }
+    }
+}
+
+void PowerSupply::setStandbyState(bool standbyState)
+{
+    _standbyState = standbyState;
+    if(fuseState)
+    {
+        if (_standbyState)
+            mainOutputVoltage = mainsInputVoltage * powerTransformerMainRatio;
+        else
+            mainOutputVoltage = 0.f;
+    }
+}
+
+void PowerSupply::blowFuse()
+{
+    fuseState = false;
+    mainOutputVoltage = 0;
+    heaterOutputVoltage = 0;
+}
+
 struct OutputSection
 {
-    // 1) the number of output tubes (int)
     int numberOfOutputTubes = 2;
-    // 2) the grid bias voltage (float)
     float gridBiasVoltage = -50.f;
-    // 3) the output transformers primary impedance in Ohms (double)
     double outputTransformerPrimaryImpedance { 2600 };
-    // 4) the output transformers secondary impedance in Ohms (float)
     float outputTransformerSecondaryImpedance = 8.f;
-    // 5) the maximum output power in watts (float)
     float maxOutputPower = 50.f;
 
-    // 1) Warm up tubes
     void warmUpTubes(float timeInMSTillWarm);
-    // 2) Admust master output volume
     void adjustMasterOutputVolume(float newVolume);
-    // 3) Amplify line level audio to a speaker level signal
     float amplifyLineLevelAudioToSpeakerLevel(float inputSignal);
+
+private:
+    bool _tubesAreWarm;
+    float _masterOutputGain;
+    float _gainFactor = numberOfOutputTubes * 24.f;
+
```

Comment:
don't use `private`.  private is covered in Project 7
Don't use underscores in member variable names
- [ ] revise